mslice/plotting/plot_window/plot_figure.py

```diff
@@ -48,6 +52,74 @@ def is_slice_figure(self):
     def is_cut_figure(self):
         return not bool(self.canvas.figure.gca().get_images())
 
+    def add_slice_plotter(self, slice_plotter):
+        self.slice_plotter = slice_plotter
+        self.menuIntensity.setDisabled(False)
+        self.ws_title = self.title
+        self.actionS_Q_E.triggered.connect(partial(self.show_intensity_plot, self.actionS_Q_E,
+                                                   self.slice_plotter.show_scattering_function, False))
+        self.actionChi_Q_E.triggered.connect(partial(self.show_intensity_plot, self.actionChi_Q_E,
+                                                     self.slice_plotter.show_dynamical_susceptibility, True))
+        self.actionChi_Q_E_magnetic.triggered.connect(partial(self.show_intensity_plot, self.actionChi_Q_E_magnetic,
+                                                              self.slice_plotter.show_dynamical_susceptibility_magnetic,
+                                                              True))
+
+    def intensity_selection(self, selected):
+        '''Ticks selected and un-ticks other intensity options. Returns previous selection'''
+        options = self.menuIntensity.actions()
+        previous = None
+        for op in options:
+            if op.isChecked() and op is not selected:
+                previous = op
+            op.setChecked(False)
+        selected.setChecked(True)
+        return previous
+
+    def show_intensity_plot(self, action, slice_plotter_method, temp_dependent):
+        if action.isChecked():
+            previous = self.intensity_selection(action)
+            cbar_log = self.colorbar_log
+            x_range = self.x_range
+            y_range = self.y_range
+            title = self.title
+            if temp_dependent:
+                if self.temp_handler(slice_plotter_method, previous) is 'cancelled':
+                    return
+            else:
+                slice_plotter_method(self.ws_title)
+            self.change_slice_plot(self.colorbar_range, cbar_log)
+            self.x_range = x_range
+            self.y_range = y_range
+            self.title = title
+            self.canvas.draw()
+        else:
+            action.setChecked(True)
+
+    def temp_handler(self, slice_plotter_method, previous):
+            try:
+                slice_plotter_method(self.ws_title)
+            except ValueError:  # sample temperature not yet set
+                try:
+                    field = self.ask_sample_temperature_field(str(self.ws_title))
+                except RuntimeError:  # if cancel is clicked, go back to previous selection
+                    self.intensity_selection(previous)
+                    return 'cancelled'
```

Comment:
I think using a string like this could make this a little brittle as the string is defined in 2 places. How about just using `True`/`False` return values for the caller to represent whether the operation was a success?